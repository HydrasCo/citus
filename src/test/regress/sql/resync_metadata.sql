CREATE SCHEMA resync_metadata;
SET search_path TO resync_metadata;

CREATE TABLE test_serial(a bigserial PRIMARY KEY);
SELECT create_distributed_table('test_serial', 'a');

CREATE SEQUENCE myseq;
CREATE TABLE test_sequence(a bigint DEFAULT nextval('myseq'));
SELECT create_distributed_table('test_sequence', 'a');

\c - - - :worker_1_port
SET search_path tO resync_metadata;

INSERT INTO test_serial VALUES(DEFAULT) RETURNING *;
INSERT INTO test_serial VALUES(DEFAULT) RETURNING *;

INSERT into test_sequence VALUES(DEFAULT) RETURNING *;
INSERT into test_sequence VALUES(DEFAULT) RETURNING *;

\c - - - :master_port
SELECT start_metadata_sync_to_node('localhost', :worker_1_port);


\c - - - :worker_1_port
SET search_path tO resync_metadata;

-- can continue inserting with the existing sequence/serial
INSERT INTO test_serial VALUES(DEFAULT) RETURNING *;
INSERT INTO test_serial VALUES(DEFAULT) RETURNING *;

INSERT into test_sequence VALUES(DEFAULT) RETURNING *;
INSERT into test_sequence VALUES(DEFAULT) RETURNING *;

\c - - - :master_port
SET search_path tO resync_metadata;
SET client_min_messages TO ERROR;
DROP SCHEMA resync_metadata CASCADE;
