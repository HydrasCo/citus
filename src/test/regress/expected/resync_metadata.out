CREATE SCHEMA resync_metadata;
SET search_path TO resync_metadata;
CREATE TABLE test_serial(a bigserial PRIMARY KEY);
SELECT create_distributed_table('test_serial', 'a');
 create_distributed_table
---------------------------------------------------------------------

(1 row)

CREATE SEQUENCE myseq;
CREATE TABLE test_sequence(a bigint DEFAULT nextval('myseq'));
SELECT create_distributed_table('test_sequence', 'a');
 create_distributed_table
---------------------------------------------------------------------

(1 row)

\c - - - :worker_1_port
SET search_path tO resync_metadata;
INSERT INTO test_serial VALUES(DEFAULT) RETURNING *;
        a
---------------------------------------------------------------------
 3940649673949185
(1 row)

INSERT INTO test_serial VALUES(DEFAULT) RETURNING *;
        a
---------------------------------------------------------------------
 3940649673949186
(1 row)

INSERT into test_sequence VALUES(DEFAULT) RETURNING *;
        a
---------------------------------------------------------------------
 3940649673949185
(1 row)

INSERT into test_sequence VALUES(DEFAULT) RETURNING *;
        a
---------------------------------------------------------------------
 3940649673949186
(1 row)

\c - - - :master_port
SELECT start_metadata_sync_to_node('localhost', :worker_1_port);
 start_metadata_sync_to_node
---------------------------------------------------------------------

(1 row)

\c - - - :worker_1_port
SET search_path tO resync_metadata;
-- can continue inserting with the existing sequence/serial
INSERT INTO test_serial VALUES(DEFAULT) RETURNING *;
        a
---------------------------------------------------------------------
 3940649673949187
(1 row)

INSERT INTO test_serial VALUES(DEFAULT) RETURNING *;
        a
---------------------------------------------------------------------
 3940649673949188
(1 row)

INSERT into test_sequence VALUES(DEFAULT) RETURNING *;
        a
---------------------------------------------------------------------
 3940649673949187
(1 row)

INSERT into test_sequence VALUES(DEFAULT) RETURNING *;
        a
---------------------------------------------------------------------
 3940649673949188
(1 row)

\c - - - :master_port
SET search_path tO resync_metadata;
SET client_min_messages TO ERROR;
DROP SCHEMA resync_metadata CASCADE;
